### DISCUSSION TITLE
How to fix "LOG packages drop detected"?

### DISCUSSION THREAD

**Discussion by cc299792458** [bitcraze/discussions#615] [Q&A] [answered]
Hi, folks

I meet a problem that I use python script to send command to crazyflie but it seems like getting stuck.

There are 3 commands in total. First command is set controller to pid controller(1), second command is set controller to a customized controller(4). The last command is using a set_point command to give set point to crazyflie.

But nothing happen when I was doing this. The most strange thing is, when I ran the exact same script last night, it worked well. However, when I run it this afternoon, it just looks like getting stuck.

I use crazyflie.console to print console information, and it recursively reports that "LOG packages drop detected", so I guess that must be the reason.

May I ask how to fix this problem?

Thanks in advance!

**Comment by cc299792458** [comment, marked as answer]
Hi, I try to comment some of my LogConfig and it can work. So I guess maybe I log too much variables and that cause the stuck?

**Comment by knmcguire** [comment]
Hi! could you perhaps share some of the code you have written? Perhaps something goes wrong with the initialization.

**Reply by cc299792458** [reply]
Sorry for replying late because a little busy!

Sure I can share my code! I'd like to share with you the code of my "crazyflie class" called "Marvel". There are some customized functions containing name "gimbal". Others are some common functions.

If you have other questions, just reply to let me know! I really appeciate your help.

![Screenshot from 2023-03-20 16-59-50](https://user-images.githubusercontent.com/52767591/226293115-209aa735-e627-482b-bdf4-4e8d5bb2833a.png)
![Screenshot from 2023-03-20 17-00-02](https://user-images.githubusercontent.com/52767591/226293150-fa0d0f78-d657-4cbc-9865-7f33fa2ea600.png)
![Screenshot from 2023-03-20 17-00-13](https://user-images.githubusercontent.com/52767591/226293162-a90560b0-5274-4d21-9b7c-23e4346f4ed3.png)
![Screenshot from 2023-03-20 17-00-24](https://user-images.githubusercontent.com/52767591/226293176-0ecd4f7b-0879-4634-bc2c-9480c8d7f482.png)
![Screenshot from 2023-03-20 17-00-38](https://user-images.githubusercontent.com/52767591/226293187-c628d1fc-fefb-4002-8c57-1685ae22f079.png)
![Screenshot from 2023-03-20 17-00-48](https://user-images.githubusercontent.com/52767591/226293204-b22b6590-c1df-4314-9357-39dd945525bb.png)

**Reply by krichardsson** [reply]
Thanks for the code!
That is a lot of code though, a bit too much, we simply do not have the time to dig though it! I suggest that you start to remove parts, piece by piece, and try to find out what is causing the problem. Try to distill the problem down to a few lines of code to make it more manageable. 

Also, please share code as text, as opposed to images. Images are not searchable and hard to work with. You can read about [how to format code in the github documentation](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks)

**Reply by cc299792458** [reply]
Thanks for the instructions. I still have another question. Is there a recommended maximum number of logging variables? In my practical experience, I found when I logged too many variables, the crazyflie seemed getting stuck. When I reduce the number of it (like from 12 to 8 or 6), it became fine.

**Comment by jackyyy65** [comment]
The solution to the problem of getting stuck while sending commands to Crazyflie drone is to troubleshoot the issue related to the loss of data packages. This can be done by checking the connection, reviewing the Python script, and resetting the Crazyflie drone.

**Reply by hmllr** [reply]
This answer seems to be generated with a tool like ChatGPT - even though there might be some truth in it, those answers will not help much in solving the problem. Please do not answer in our discussions with ChatGPT, it will just generate noise and make the real and useful answers harder to find.
